File: gitnexus/test/fixtures/analysis-quality/typescript-class-inheritance-basic/src/dog.ts
import { Animal } from './animal';

export class Dog extends Animal {
  breed: string;

  constructor(name: string, breed: string) {
    super(name);
    this.breed = breed;
  }

  speak(): string {
    return `${this.name} barks.`;
  }

  fetch(): string {
    return `${this.name} fetches the ball.`;
  }
}

File: gitnexus/test/fixtures/lang-resolution/objc-calls/Dog.m
// Dog.m — Dog implementation with cross-file message calls
#import "Dog.h"

@implementation Dog

- (NSString *)speak {
    return @"Woof!";
}

- (void)fetch:(NSString *)item {
    // triggers CALLS edge: [self eat] → Animal.eat
    [self eat];
    NSLog(@"%@ fetched %@", self.name, item);
}

@end
